walinfo: "Invalid" partial WAL file

walinfo can't parse the attached file - pg_waldump can

I've seen this problem before but I couldn't solve it back then.
```
➜  ~ pgmoneta-walinfo ~/Downloads/00000001000000000000000E.partial.gz
Error while reading/describing WAL file
➜  ~ cat /tmp/pgmoneta.log
2025-07-13 17:12:31 TRACE utils.c:2029 FILETRACKER | Copy | /home/mohab/Downloads/00000001000000000000000E.partial.gz | /tmp |
2025-07-13 17:12:31 FATAL wal_reader.c:604 BKPBLOCK_HAS_DATA not set, but data length is not zero
2025-07-13 17:12:31 FATAL wal_reader.c:503 Error: Could not parse WAL file
2025-07-13 17:12:31 ERROR walfile.c:111 Failed to parse WAL file: /tmp/00000001000000000000000E.partial
2025-07-13 17:12:31 FATAL walfile.c:383 Failed to read WAL file at /home/mohab/Downloads/00000001000000000000000E.partial.gz
➜  ~ 
```

I'll do my best.

--- a/src/libpgmoneta/walfile/wal_reader.c
+++ b/src/libpgmoneta/walfile/wal_reader.c
@@ -232,11 +232,7 @@ pgmoneta_wal_parse_wal_file(char* path, int server, struct walfile* wal_file)
       server_config = &config->common.servers[server];
    }
 
-   uint32_t next_record = MAXALIGN(
-      ftell(file) +
-      ((long_header->std.xlp_rem_len / long_header->xlp_xlog_blcksz) * SIZE_OF_XLOG_SHORT_PHD) +
-      long_header->std.xlp_rem_len % long_header->xlp_xlog_blcksz
-      );
+   uint32_t next_record;
    int page_number = 0;
    wal_file->long_phd = long_header;
    read_all_page_headers(file, wal_file->long_phd, wal_file);
@@ -250,6 +246,12 @@ pgmoneta_wal_parse_wal_file(char* path, int server, struct walfile* wal_file)
 
    if (long_header->std.xlp_rem_len > 0)
    {
+      next_record = MAXALIGN(
+         ftell(file) +
+         ((long_header->std.xlp_rem_len / long_header->xlp_xlog_blcksz) * SIZE_OF_XLOG_SHORT_PHD) +
+         long_header->std.xlp_rem_len % long_header->xlp_xlog_blcksz
+         );
+
       if (partial_record->xlog_record_bytes_read == 0)
       {
          decoded = malloc(sizeof(struct decoded_xlog_record));
@@ -272,6 +274,10 @@ pgmoneta_wal_parse_wal_file(char* path, int server, struct walfile* wal_file)
          temp_next_record = next_record;
          next_record = SIZE_OF_XLOG_LONG_PHD;
       }
+   }
+   else
+   {
+      next_record = long_header->xlp_xlog_blcksz;
    }
    long_header = NULL;
    fseek(file, next_record, SEEK_SET);

